ansible/roles/test/files/ptftests/hash_test.py

```diff
@@ -189,33 +193,37 @@ def check_within_expected_range(self, actual, expected):
         return (percentage, abs(percentage) <= self.balancing_range)
 
     #---------------------------------------------------------------------
-    def check_balancing(self, dest_port_list, port_hit_cnt):
+    def check_balancing(self, dest_port_list, port_hit_cnt, mode='strict'):
         '''
         @summary: Check if the traffic is balanced across the ECMP groups and the LAG members
         @param dest_port_list : a list of ECMP entries and in each ECMP entry a list of ports
         @param port_hit_cnt : a dict that records the number of packets each port received
+        @param mode : check mode, strict or loose, default for strict
         @return bool
         '''
 
         logging.info("%-10s \t %-10s \t %10s \t %10s \t %10s" % ("type", "port(s)", "exp_cnt", "act_cnt", "diff(%)"))
         result = True
 
-        total_hit_cnt = sum(port_hit_cnt.values())
-        for ecmp_entry in dest_port_list:
-            total_entry_hit_cnt = 0
-            for member in ecmp_entry:
-                total_entry_hit_cnt += port_hit_cnt.get(member, 0)
-            (p, r) = self.check_within_expected_range(total_entry_hit_cnt, float(total_hit_cnt)/len(dest_port_list))
-            logging.info("%-10s \t %-10s \t %10d \t %10d \t %10s"
-                        % ("ECMP", str(ecmp_entry), total_hit_cnt/len(dest_port_list), total_entry_hit_cnt, str(round(p, 4)*100) + '%'))
-            result &= r
-            if len(ecmp_entry) == 1 or total_entry_hit_cnt == 0:
-                continue
-            for member in ecmp_entry:
-                (p, r) = self.check_within_expected_range(port_hit_cnt.get(member, 0), float(total_entry_hit_cnt)/len(ecmp_entry))
+        if mode == 'loose':
+            result = True if len(port_hit_cnt.keys()) > 1 else False
```

Comment:
even in loose mode, should still print out the hit_map.

Reply:
there is another loose mode in which at least every egress port is hitted once, which makes more sense to me.

Reply:
> even in loose mode, should still print out the hit_map for debugging purpose.

For the changes in the last two review comments, I delete the loose mode. And the logging info "logging.info("hit count map: {}".format(hit_count_map))" is enough.